### PR TITLE
*: add ColumnLimit for clang-format

### PR DESCRIPTION
**Pull request by gengliqi**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/7896

Problem Summary:
I choose a few files to show the effect of changing the clang-format.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Flash/executeQuery.cpp [hunk 1]
```diff
@@ -52,7 +52,8 @@ void prepareForExecute(Context & context)
     context.setQueryContext(context);
 
     QuotaForIntervals & quota = context.getQuota();
-    quota.addQuery(); /// NOTE Seems that when new time interval has come, first query is not accounted in number of queries.
+    quota
+        .addQuery(); /// NOTE Seems that when new time interval has come, first query is not accounted in number of queries.
```
Comment (Lloyd-Pottiger) [inline dbms/src/Flash/executeQuery.cpp:56]: looks not reasonable, can we do not count comment into line length?

File: dbms/src/Interpreters/Settings.h [hunk 2]
```diff
@@ -39,282 +39,729 @@ struct Settings
...
-    M(SettingUInt64, recv_queue_size, 0, "size of ExchangeReceiver queue, 0 means the size is set to data_source_mpp_task_num * 50")                                                                                                    \
-    M(SettingUInt64, shallow_copy_cross_probe_threshold, 0, "minimum right rows to use shallow copy probe mode for cross join, default is max(1, max_block_size/10)")                                                                   \
-    M(SettingInt64, max_buffered_bytes_in_executor, 200LL * 1024 * 1024, "The max buffered size in each executor, 0 mean unlimited, use 200MB as the default value")                                                                    \
-    M(SettingUInt64, ddl_sync_interval_seconds, 60, "The interval of background DDL sync schema in seconds")                                                                                                                            \
+#define APPLY_FOR_SETTINGS(M)                                                                                          \
```
Comment (JaySon-Huang) [inline dbms/src/Interpreters/Settings.h:42]: I think keep `clang-format off` on this file is more reasonable

File: dbms/src/Flash/LogSearch.h [hunk 3]
```diff
@@ -139,9 +139,7 @@ class LogIterator : private boost::noncopyable
 void ReadLogFile(const std::string & path, std::function<void(std::istream &)> && cb);
 
 bool FilterFileByDatetime(
-    const std::string & path,
-    const std::vector<std::string> & ignore_log_file_prefixes,
-    int64_t start_time);
+    const std::string & path, const std::vector<std::string> & ignore_log_file_prefixes, int64_t start_time);
```
Comment (JaySon-Huang) [inline dbms/src/Flash/LogSearch.h:142]: I think split it into multiple lines is more reasonable

File: dbms/src/Flash/CoprocessorHandler.h [hunk 4]
```diff
@@ -37,7 +37,8 @@ struct CoprocessorContext
     const kvrpcpb::Context & kv_context;
     const grpc::ServerContext & grpc_server_context;
 
-    CoprocessorContext(Context & db_context_, const kvrpcpb::Context & kv_context_, const grpc::ServerContext & grpc_server_context_);
+    CoprocessorContext(
+        Context & db_context_, const kvrpcpb::Context & kv_context_, const grpc::ServerContext & grpc_server_context_);
```
Comment (JaySon-Huang) [inline dbms/src/Flash/CoprocessorHandler.h:41]: Is it caused by `ColumnLimit`? This method declares all params in the following line.
I prefer the `CoprocessorHandler` style, which declares params in multiple lines.

I prefer:
* For the short one, the function name and its params are in the same line
* For the long one, the function name and all its params are in different lines

Comment (gengliqi) [reply dbms/src/Flash/CoprocessorHandler.h]: Got it. I change the `AllowAllArgumentsOnNextLine` and `AllowAllParametersOfDeclarationOnNextLine` back to false.

File: dbms/src/Interpreters/Settings.h (hunk 2 above)
Comment (gengliqi) [reply dbms/src/Interpreters/Settings.h:42]: Done.

File: dbms/src/Flash/executeQuery.cpp (hunk 1 above)
Comment (gengliqi) [reply dbms/src/Flash/executeQuery.cpp:56]: I search for this issue. Unfortunately, it's no way to make `clang-format` ignore the length of comments.
Only two things we can do:
1. turn `clang-format` off for this code.
2. set `ReflowComments` to true then this line of code will be look like
```
quota.addQuery(); /// NOTE Seems that when new time interval has come, first query is not accounted in number of
                  /// queries.
```
Seems ok. But I don't know if it's good enough for all the other comments.

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**